### PR TITLE
[ELY-227] - Attribute mapping to the Jdbc Security Realm.

### PR DESCRIPTION
**Pull request by pedroigor**
It depends on https://github.com/wildfly-security/wildfly-elytron/pull/224. It should be merged first.

Thanks.

### CODE REVIEW

**Comment by pedroigor** [comment]
ELY-SUB PR: https://github.com/wildfly-security/elytron-subsystem/pull/93

**Comment by pedroigor** [comment]
Rebased.

**Comment by pedroigor** [comment]
Rebased

File: src/main/java/org/wildfly/security/auth/provider/jdbc/JdbcSecurityRealm.java [hunk 1]
```diff
@@ -111,13 +108,9 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) throws Re
...
-                for (ColumnMapper mapper : configuration.getColumnMappers()) {
-                    if (KeyMapper.class.isInstance(mapper)) {
-                        KeyMapper keyMapper = (KeyMapper) mapper;
-
-                        if (keyMapper.getKeyType().isAssignableFrom(credentialType)) {
-                            return executeAuthenticationQuery(configuration, resultSet -> (C) keyMapper.map(resultSet));
-                        }
+                for (KeyMapper keyMapper : configuration.getColumnMappers(KeyMapper.class)) {
+                    if (keyMapper.getKeyType().isAssignableFrom(credentialType)) {
+                        return executePrincipalQuery(configuration, resultSet -> (C) keyMapper.map(resultSet));
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/auth/provider/jdbc/JdbcSecurityRealm.java]: I missed this earlier, but this looks like an unsafe cast.  You could make it safe by replacing ``(C) blah`` with ``credentialType.cast(blah)``.

File: src/main/java/org/wildfly/security/auth/provider/jdbc/QueryConfiguration.java [hunk 2]
```diff
@@ -62,7 +67,19 @@ public DataSource getDataSource() {
...
+    List<ColumnMapper> getColumnMappers() {
+        return Collections.unmodifiableList(this.columnMappers);
+    }
+
+    <T extends ColumnMapper> List<T> getColumnMappers(Class<T> mapperType) {
+        List<T> attributeMappers = new ArrayList<>();
+
+        for (ColumnMapper columnMapper : this.columnMappers) {
+            if (mapperType.isInstance(columnMapper)) {
+                attributeMappers.add((T) columnMapper);
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/auth/provider/jdbc/QueryConfiguration.java]: Here's another unsafe cast that can be changed to use ``mapperType.cast()`` instead.

**Pull request merged**